lib/WatchmanFileCache.js

```diff
@@ -0,0 +1,122 @@
+import fbWatchman from 'fb-watchman';
+
+const SUBSCRIPTION_NAME = 'import-js-subscription';
+
+const CACHE = new Set();
+
+// Flag to quickly check if the cache is enabled.
+let enabled = false;
+
+function subscribe({ client, fbWatch, relativePath }): Promise {
+  const subscription = {
+    // Match any `.js` file
+    expression: ['allof', ['match', '*.js*']],
```

Comment:
I've played around with `type` now, and I can't get it to work for symlinks. And having directories end up in the cache isn't really a problem, since we apply a fairly sophisticated regexp to grep for the "right" files. And the other finders are not filtering out directories either, so doing so here would make things inconsistent. I'm going to leave this untouched.